xcell/mappers/mapper_KiDS1000.py

```diff
@@ -57,40 +54,27 @@ def __init__(self, config):
 
     def get_catalog(self):
         if self.cat_data is None:
-            read_lite, fname_lite = self._check_lite_exists(self.zbin, 'cat')
-            if read_lite:
-                print(f'loading lite cat {self.zbin}', flush=True)
-                self.cat_data = Table.read(fname_lite, format='fits')
-            else:
-                print(f'loading full cat {self.zbin}', flush=True)
-                self.cat_data = self._load_catalog()
-
-            print('Catalogs loaded', flush=True)
-
+            fn = f'KiDS1000_cat_bin{self.zbin}.fits'
+            self.cat_data = self._rerun_read_cycle(fn, 'FITSTable',
+                                                   self._load_catalog,
+                                                   already_saved=True)
         return self.cat_data
 
     def _load_catalog(self):
         nzbins = self.zbin_edges.shape[0]
-        data = []
-
-        data_cat = Table.read(self.config['data_catalog'],
+        cats = []
+        cat_full = Table.read(self.config['data_catalog'],
                               format='fits')[self.column_names]
-        for i in range(nzbins):
-            # binning
-            sel = self._bin_z(data_cat, i)
-            data_zbin = data_cat[sel]
-            self._remove_additive_bias(data_zbin)
-            data.append(data_zbin)
-
-        for zbin, cat_zbin in enumerate(data):
-            self._remove_multiplicative_bias(cat_zbin, zbin)
-            read_lite, fname_lite = self._check_lite_exists(zbin, 'cat')
-            print(fname_lite)
-            if fname_lite is not None:
-                print(f'Writing lite catalog: {fname_lite}', flush=True)
-                cat_zbin.write(fname_lite)
-
-        return data[self.zbin]
+        for ibin in range(nzbins):
+            sel = self._bin_z(cat_full, ibin)
+            cat = cat_full[sel]
+            self._remove_additive_bias(cat)
+            self._remove_multiplicative_bias(cat, ibin)
+            cat = cat.as_array()
+            fn = f'KiDS1000_cat_bin{ibin}.fits'
+            save_rerun_data(self, fn, 'FITSTable', cat)
+            cats.append(cat)
+        return cats[self.zbin]
```

Comment:
Since it will only return one, it might make more sense just to keep the catalog of the zbin of interest. That will save memory. I'd add in the for loop an `if ibin == self.zbin; cat_to_return = cat` and then `return cat_to_return` when it finishes.

Reply:
done